scipy/sparse/linalg/interface.py

```diff
@@ -325,7 +327,7 @@ def matmat(self, X):
             raise ValueError('expected 2-d ndarray or matrix, not %d-d'
                              % X.ndim)
 
-        M,N = self.shape
+        _, N = self.shape
```

Comment:
Would rather see `N = self.shape[1]` here.

Reply:
Done - I just get rid of N